src/test/osd/osdcap.cc

```diff
@@ -270,6 +270,11 @@ TEST(OSDCap, BasicRW) {
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", false, true, false, true));
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", true, false, true, false));
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", true, true, true, true));
+
+  // class whitelist flag works
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", false, true, false, true, true));
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", true, false, true, false, true));
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", true, true, true, true, true));
```

Comment:
My understanding was that if a class was whitelisted, then irregardless of having 'allow x' it should be permitted to call into an object class. So this test is supposed to show that with only 'allow rw' but a whitelisted class (last parameter: true) then may_class_read and may_class_write are OK. Have I got these semantics messed up?